pkg/operator/controllers/storageaccounts/storageaccount_controller.go

```diff
@@ -116,11 +115,15 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 	aroClusterPredicate := predicate.NewPredicateFuncs(func(o client.Object) bool {
 		return o.GetName() == arov1alpha1.SingletonClusterName
 	})
+	masterMachinePredicate := predicate.NewPredicateFuncs(func(o client.Object) bool {
+		role, ok := o.GetLabels()["machine.openshift.io/cluster-api-machine-role"]
+		return ok && role == "master"
+	})
 
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&arov1alpha1.Cluster{}, builder.WithPredicates(aroClusterPredicate)).
-		Watches(&source.Kind{Type: &machinev1beta1.Machine{}}, &handler.EnqueueRequestForObject{}). // to reconcile on machine replacement
-		Watches(&source.Kind{Type: &corev1.Node{}}, &handler.EnqueueRequestForObject{}).            // to reconcile on node status change
+		Watches(&source.Kind{Type: &machinev1beta1.Machine{}}, &handler.EnqueueRequestForObject{}, builder.WithPredicates(masterMachinePredicate)). // to reconcile on master machine replacement
```

Comment:
What would be the impact of putting in a `Watches` for control plane machinesets now (before we have them in many/most ARO clusters) so that we don't have to go and change that later? I'm assuming the end goal is to only watch machinesets and control plane machinesets, and watching master machines is a temporary solution in the meantime.

Reply:
Assuming there are no frequent changes to the ControlPlaneMachineSet resources, I don't imagine there would be a wider negative impact. However, the code to actually extract the in-use cluster subnets will, due to the timing you called out, only be reading master Machines and worker MachineSets anyways, so we'd be adding a Watch to a resource we never actually read. 

I think if we get to a point where we only ever watch MachineSets and ControlPlaneMachineSets, we'd need a similar PR to this one to update both our Watches and our actual subnet retrieval code. 

I'm personally in favor of leaving this as-is, especially since we've discussed internally about the possibility of merging these two controllers into a single one anyways (to cut down on duplicate GET Subnet Azure calls).